### ISSUE TITLE
Option without Insightface

### ISSUE THREAD

**Issue by theZaX** [fudan-generative-vision/hallo#142]
Incredible work! This is an excellent project.

Wanted to see if using something other than Insightface would be possible and if you could describe how such a change could be made. Happy to help with some of the dev work around that if possibe.

Insightface is non-commercial on their models and has a history of abusing the youtube copyright system.

**Comment by subazinga** [comment]
Indeed, alternative methods can replace it since it's used solely to obtain the mask of the lips and face. However, the performance impact is uncertain because our training data was constructed based on InsightFace.

**Comment by theZaX** [comment]
Interesting makes sense, could the model be retrained using the training code given? How much training data would be required?
